test/integration/BUILD

```diff
@@ -33,3 +50,25 @@ go_test(
         "//vendor/k8s.io/client-go/kubernetes:go_default_library",
     ],
 )
+
+genrule(
+    name = "integration-provider-components",
+    srcs = [
+        "//test:kustomization.yaml",
+        "//test:manager-version-patch.yaml",
+        "//config:aws_provider.yaml",
+    ],
+    outs = ["manifests/provider-components-integration.yaml"],
+    cmd = """cp -f {kustomization} $(@D) && \\
```

Comment:
The `join` based ones were Liz's, whereas I independently started doing the format based ones and I've preserved both in this PR rather than refactoring them all to one type.

Wasn't sure what the consensus was, so can do a follow up to standardise.

Reply:
Switched to `join` based one in this PR.